sqliterepo/sqlite_utils.c

```diff
@@ -594,3 +594,38 @@ db_properties_to_json(
 	g_string_append_static(json, "}");
 	return json;
 }
+
+static int64_t _sqlx_get_number(struct sqlx_sqlite3_s *sq3, const gchar *req) {
+	int result = -1;
+	gint rc;
+	sqlite3_stmt *stmt = NULL;
+
+	sqlite3_prepare_debug(rc, sq3->db, req, -1, &stmt, NULL);
+	if (rc == SQLITE_OK) {
+		EXTRA_ASSERT(stmt != NULL);
+		if (SQLITE_ROW == (rc = sqlite3_step(stmt))) {
+			result = sqlite3_column_int64(stmt, 0);
+		}
+	}
+	sqlite3_finalize(stmt);
+	return result;
+}
+
+GPtrArray* sqlx_admin_get_usage(struct sqlx_sqlite3_s *sq3) {
+	GPtrArray *result = g_ptr_array_new();
+	char *tmp = NULL;
+
+	g_ptr_array_add(result, g_strdup("stats.page_count"));
+	asprintf(&tmp, "%ld", _sqlx_get_number(sq3, "PRAGMA main.page_count"));
```

Comment:
Danger ! Danger ! Danger !
`asprintf` is a GNU C library function, whereas `g_ptr_array_add` (and all other functions we use) are GLib functions. GLib has its own memory management system.
Use `g_strdup_printf("%"G_GINT64_FORMAT, ...)` instead.